### DISCUSSION TITLE
Configuration to prevent ghasts from using portals

### DISCUSSION THREAD

**Discussion by CatTeaA** [PurpurMC/Purpur#1378] [Feature Requests]
### Spark link

https://spark.lucko.me/ooh5SoyreL

### Description of issue

Ghasts Teleporting through Nether Portals cause significantly more lag than other Entities.
This should be a matter of vanilla

When a Ghast uses a portal, it causes a lag spike on the server.
It would be nice if Purpur had a configuration that prevented Ghasts from using portals, or had this lag fix patch.
https://github.com/pufferfish-gg/Pufferfish/issues/76
https://github.com/PaperMC/Paper/issues/9173

### Purpur version

2006-1.20.1

### Agreements

- [X] I am running the latest version of Purpur available from https://purpurmc.org/downloads.
- [X] I have searched for and ensured there isn't already an open issue regarding this.
- [ ] I ticked all the boxes without actually reading them
- [X] My version of Minecraft is supported by Purpur.

**Comment by Potothingi** [comment]
Good. Purpur has the 'entities-can-use-portals' configuration, so I think it will be possible with a small modification to this patch.
https://github.com/PurpurMC/Purpur/blob/ver/1.20.1/patches/server/0088-Entities-can-use-portals-configuration.patch

**Comment by CatTeaA** [comment]
Issue with Ghast's collision box size
Is it possible to modify the Ghast's collision box size by adding an option，to solve this vanilla lag problem
like this:
https://purpurmc.org/docs/Configuration/#short-height
![239664140-e77bdd65-a8b9-4c38-a8fa-9fa20e9dbd62](https://github.com/PurpurMC/Purpur/assets/135555687/845ca09c-7210-4a16-a320-d829d9c3f962)

**Comment by Potothingi** [comment]
I want to disable the portal only for Ghast and not all entities.

Wouldn't it be possible to change it like below? (I don't know how to change git diff HEAD.)
https://github.com/Potothingi/Purpur/commit/423070ffe5ca874c6c7d753db0e559bad8a469c6

**Comment by Potothingi** [comment]
I think the best solution is to add a configuration that skips the collision box size check if the entity moving through the portal is a Ghast.
